src/main/java/org/openrewrite/java/testing/junit5/AssertThrowsOnLastStatement.java

```diff
@@ -113,29 +112,103 @@ public J.MethodDeclaration visitMethodDeclaration(J.MethodDeclaration methodDecl
+
+                        Object type = "Object";
+                        if (e.getType() instanceof JavaType.Primitive) {
+                            type = e.getType().toString();
+                        } else if (e.getType() != null && TypeUtils.asClass(e.getType()) != null) {
+                            type = TypeUtils.asClass(e.getType()).getClassName();
+                            maybeAddImport(TypeUtils.asFullyQualified(e.getType()).getFullyQualifiedName(), false);
+                        }
+
+                        J.VariableDeclarations varDecl = new J.VariableDeclarations(Tree.randomId(),
+                                Space.EMPTY,
+                                Markers.EMPTY,
+                                Collections.emptyList(),
+                                Collections.emptyList(),
+                                new J.Identifier(Tree.randomId(),
+                                        Space.EMPTY,
+                                        Markers.EMPTY,
+                                        Collections.emptyList(),
+                                        type.toString(),
+                                        e.getType(),
+                                        null),
+                                null,
+                                Collections.emptyList(),
+                                Arrays.asList(JRightPadded.build(new J.VariableDeclarations.NamedVariable(Tree.randomId(),
+                                        Space.SINGLE_SPACE,
+                                        Markers.EMPTY,
+                                        new J.Identifier(Tree.randomId(),
+                                                Space.EMPTY,
+                                                Markers.EMPTY,
+                                                Collections.emptyList(),
+                                                getVariableName(e),
+                                                e.getType(),
+                                                new JavaType.Variable(null, 0, getVariableName(e), null, e.getType(), null)),
+                                        Collections.emptyList(),
+                                        JLeftPadded.build((Expression)e.withPrefix(Space.SINGLE_SPACE)).withBefore(Space.SINGLE_SPACE),
+                                        new JavaType.Variable(null, 0, getVariableName(e), null, e.getType(), null)))));
+
+                        J.Identifier name = varDecl.getVariables().get(0).getName();
+                        statements.add(varDecl.withPrefix(methodStatement.getPrefix().withComments(emptyList())));
+                        lambdaArguments.add(name);
+                    }
+                    lambdaStatement = mi.withArguments(lambdaArguments);
+                }
+                return lambdaStatement;
+            }
```

Comment:
as [discussed on Slack](https://moderneinc.slack.com/archives/C088P8067NH/p1746806201767849?thread_ts=1746715345.295159&cid=C088P8067NH) this was based in part on an example that could be simplified with an anonymous visitor; putting that reminder here to use when reviewing & polishing, as I don't expect you to know what adjustments to make based on that comment alone.

Reply:
Good call, I'm going to review this one after the code has been simplified.